### PR TITLE
Make contact pressure a generic variable for all contact pairs

### PR DESCRIPTION
**Pull request by recuero**
If names are not distinct, these will be overridden and only one of the contact pairs will have the correct contact pressure or penetration information. This will require regolding tests due to the name change. 

 (#18022)

### CODE REVIEW

**Comment by recuero** [comment]
This PR requires downstream updates.

File: modules/contact/test/tests/multiple_contact_pairs/tests [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
+  design = 'syntax/Contact/index.md MechanicalContactConstraint.md ContactPressureAux.md'
   [multiple_pairs]
     type = Exodiff
     input = 'multiple_pairs.i'
     exodiff = 'multiple_pairs_out.e'
     abs_zero = 1.0e-8
     rel_err = 1.0e-5
     cli_args = "Postprocessors/active=''"
-    requirement = 'Contact module action shall allow for multiple contact pairs when selecting node-face formulation.'
+    requirement = 'Contact module action shall allow for multiple contact pairs when selecting '
```
Comment (bwspenc) [inline modules/contact/test/tests/multiple_contact_pairs/tests:11]: `'The system shall set up node-face interactions between multiple surface pairs using the Contact action'`

**Review by bwspenc — CHANGES_REQUESTED** [review]

**Comment by recuero** [comment]
Comments should have been addressed.

**Review by bwspenc — COMMENTED** [review]
All my comments have been addressed. I'm ready to approve this, but I'll hold off until we get the OK to break next.

**Review by bwspenc — APPROVED** [review]

**Comment by recuero** [comment]
```
At least 2 approving reviews are required to merge this pull request.
```

**Pull request merged**